### PR TITLE
storage: add COCKROACH_ENABLE_QUEUE_EVENT_LOG env var

### PR DESCRIPTION
**Pull request by petermattis**
Default queue eventlogs off as they are mostly filled with spam on real
clusters.

* Does anyone actually use the queue /event info outside of local debugging?

---
This change is [Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/13323)

### CODE REVIEW

**Comment by bdarnell** [comment]
LGTM. I've never really used these logs.

Should we control this with `log.V(N)` instead of a new env var? We do that occasionally for things that are not exactly normal logs.

**Comment by petermattis** [comment]
> Should we control this with log.V(N) instead of a new env var? We do that occasionally for things that are not exactly normal logs.

I'm happy to do that. There are a lot of uses of `log.{VEventf,Eventf}` in the queues. Would be good to hear how people use that logging. I for one always turn on a `--vmodule`. Anyone else have opinions?

**Comment by petermattis** [comment]
> Should we control this with log.V(N) instead of a new env var? We do that occasionally for things that are not exactly normal logs.

I added another commit which switches the `log.{VEvent,Event}` calls in `*queue.go` to `log.V(X).Info` calls. The env var is still there as it controls whether the logs also go to an event queue, though I'm not sure how useful that is for debugging. If you have the logs they are easy enough to grep. PTAL.

**Comment by bdarnell** [comment]
I think if these logs are going to be off by default we don't need two different ways of opting in to two different ways of recording them. Personally I'd get rid of the event log and just have the regular logs behind `vmodule`. I think @spencerkimball was the one who was most interested in these logs so he may want to weigh in.

**Comment by spencerkimball** [comment]
I also dislike the two ways of recording these. My thought originally was that the event logs for a context would be printed to the log as a single statement containing the full trace if the op for which the context was created ended in failure. It was not my intention that every event trace would be logged. I just don't think `vmodule` is at all useful in production. Too noisy.

I would recommend that we just eliminate the queue event logging entirely – remove the env var as well for this PR.

**Comment by petermattis** [comment]
> I would recommend that we just eliminate the queue event logging entirely – remove the env var as well for this PR.

Done. PTAL.

File: pkg/util/log/ambient_context.go [hunk 1]
```diff
@@ -99,7 +99,9 @@ func (ac *AmbientContext) SetEventLog(family, title string) {
 // FinishEventLog closes the event log. Concurrent and subsequent calls to
 // record events from contexts that use this event log embedded are allowed.
 func (ac *AmbientContext) FinishEventLog() {
-	ac.eventLog.finish()
+	if ac.eventLog != nil {
```
Comment (bdarnell) [inline pkg/util/log/ambient_context.go]: Go ahead and remove the field if we're never setting it to non-nil?

**Review by bdarnell — APPROVED** [review]

File: pkg/util/log/ambient_context.go (hunk 1 above)
Comment (petermattis) [inline pkg/util/log/ambient_context.go]: We call `SetEventLog` from gossip. I can revert this change as we now never call `FinishEventLog` without having first called `SetEventLog`.

Comment (bdarnell) [inline pkg/util/log/ambient_context.go]: Oh, nevermind. I didn't notice that this part of the PR had left the storage package.

**Pull request merged**